src/neural/cuda/network_cudnn.cc

```diff
@@ -306,34 +306,36 @@ class CudnnNetwork : public Network {
       use_custom_winograd_ = true;
     }
 
-    // Override if set in backend-opts.
-    if (!options.IsDefault<bool>("custom_winograd"))
-      use_custom_winograd_ = options.Get<bool>("custom_winograd");
-
     // Warn if the memory required for storing transformed weights is
-    // going to exceed 60% of total video memory, force custom_winograd off
-    // if it's going to exceed 80% of memory.
+    // going to exceed 40% of total video memory, force custom_winograd off
+    // if it's going to exceed 50% of memory.
     size_t residual_single_layer_weight_size =
         3 * 3 * kNumFilters * kNumFilters * sizeof(DataType);
     size_t residual_weight_size =
         residual_single_layer_weight_size * numBlocks_ * 2;
     size_t transformed_residual_weight_size = residual_weight_size * 4;
+
     if (residual_weight_size > 0.6 * deviceProp.totalGlobalMem) {
       CERR << "Low video memory detected. You may run into OOM errors. Please "
               "consider using a smaller network.";
-      // No hope of using custom winograd - even the fallback path might not run.
+    }
+    if (use_custom_winograd_ &&
+        transformed_residual_weight_size > 0.5 * deviceProp.totalGlobalMem) {
```

Comment:
Need to add `&& options.IsDefault<bool>("custom_winograd")` here to avoid this specific warning if an override is given.